data/pools/transactionPool.go

```diff
@@ -270,12 +275,22 @@ func (pool *TransactionPool) pendingCountNoLock() int {
 }
 
 // checkPendingQueueSize tests to see if we can grow the pending group transaction list
-// by adding txCount more transactions. The limits comes from the total number of transactions
+// by adding len(txnGroup) more transactions. The limits comes from the total number of transactions
 // and not from the total number of transaction groups.
 // As long as we haven't surpassed the size limit, we should be good to go.
-func (pool *TransactionPool) checkPendingQueueSize(txCount int) error {
+func (pool *TransactionPool) checkPendingQueueSize(txnGroup []transactions.SignedTxn) error {
 	pendingSize := pool.pendingTxIDsCount()
+	txCount := len(txnGroup)
 	if pendingSize+txCount > pool.txPoolMaxSize {
+		// Allow the state proof transaction to go over the txPoolMaxSize if it already didn't
+		if len(txnGroup) == 1 && txnGroup[0].Txn.Type == protocol.StateProofTx {
+			pool.pendingMu.Lock()
+			defer pool.pendingMu.Unlock()
+			if !pool.stateproofOverflowed {
+				pool.stateproofOverflowed = true
```

Comment:
So there is no situation where you would want to have more than 1 SP transaction in the tx pool if you're running behind? Looking at apply/stateproof.go, is it possible for multiple SP txns to be evaluated in the same block and each call SetStateProofNextRound in sequence to catch up?

Reply:
Limiting to 1 overflow SP per txn pool also means you will only relay one SP txn if you're running behind and your txn pool is full — you won't relay any more than that. But relays will periodically resend SP txns is the answer?

Reply:
> So there is no situation where you would want to have more than 1 SP transaction in the tx pool if you're running behind? 

What I am limiting here is in the event the transaction pool is full, and all incoming transactions are rejected, **only one** SP transaction can be admitted by "expanding" the pool.
There may or may not be other SP transactions in the pool, and after a block is created and the pool admits transactions again, more SP transactions can be added.  

> Looking at apply/stateproof.go, is it possible for multiple SP txns to be evaluated in the same block and each call SetStateProofNextRound in sequence to catch up?

In general yes. But this PR limits in the following situation:
When the transaction pool is full, and while it is full and not accepting any more transactions, two SP transactions arrive, before a new block is created, then one of them will be admitted to the pool, the other rejected. 

Note that, when the pool is full, it is likely that multiple blocks worth of transactions are already backed up. 

Now, if the rejected SP transaction is not received again, then the block will have only one SP transaction instead of 2.

This is a very rare situation, and even if it happens, it is perfectly fine behavior.

Reply:
> Limiting to 1 overflow SP per txn pool also means you will only relay one SP txn if you're running behind and your txn pool is full — you won't relay any more than that. But relays will periodically resend SP txns is the answer?

If the relay's pool is full, it will not be able to add more than one SP into its own pool and consequently won't be able to broadcast more than one SP to others. But after 1 round, will be able to.

Reply:
OK, so you're saying it's fine because relays will keep trying in this rare situation, and letting 1 in at a time per full tx pool should be sufficient to make progress. However since the txpool is ~3x the block size, if this node were a proposer it would take 3 blocks before the state proof tx makes it into a block. If this node were a relay, you might have different relays communicating to each other all with full pools, but each relay is leaving one slot open for the "next" state proof txn.

Reply:
Yes. That is what is expected. The performance evaluation on a large network is needed to confirm this.